harpy/commands/convert.py

```
@@ -340,19 +243,146 @@ def standardize(sam, quiet):
+    logtext = f"Standardizing [dim][-> [magenta]{style.lower()}[/]][/]" if style else "Standardizing"
+
+    BC_TYPE = which_linkedread(r1_fastq)
+    if not BC_TYPE:
+        print_error("undertermined file type", f"Unable to determine the linked-read barcode type after scanning the first 100 records of {os.path.basename(r1_fastq)}. Please make sure the format is one of haplotagging, stlfr, or tellseq. 10X-style with the barcode as the first 16 nucleotides of read 1 is not supported here.")
+    
+    # create the output directory in case it doesn't exist
+    if os.path.dirname(prefix):
+        os.makedirs(os.path.dirname(prefix), exist_ok=True)
+
+    if style:
+        bc_out = open(f"{prefix}.bc", "w")
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)